libs/@blockprotocol/graph/src/shared/stdlib/subgraph/edge/shared.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  BaseUrl,
+  extractBaseUrl,
+  extractVersion,
+  VersionedUrl,
+} from "@blockprotocol/type-system/slim";
+
+import {
+  OntologyOutwardEdge,
+  OntologyTypeRevisionId,
+  OntologyTypeVertexId,
+  OutwardEdge,
+  StrictOntologyOutwardEdge,
+  Subgraph,
+} from "../../../types/subgraph.js";
+
+/**
+ * This is a helper function to extract ontology edges given an `outwardEdgePredicate` function.
+ *
+ * @internal
+ * @param subgraph {Subgraph} - The `Subgraph` containing the underlying ontology types
+ * @param ontologyTypeId {OntologyTypeVertexId | VersionedUrl} - The identifier of the `EntityType` to search for
+ * @param outwardEdgePredicate {(outwardEdge: OutwardEdge<boolean>) => outwardEdge is StrictOntologyOutwardEdge} - The predicate to filter edges by
+ * @returns {OntologyTypeVertexId[]} - The resulting endpoints of the filtered edges
+ */
+export const getOntologyTypesReferencedBy = (
```

Comment:
The name of this is probably fine, I just keep wondering if we can name it to be something like `getRightEndpointsForEdgeKind` or something.. but while still maintaining that this is about '`StrictOntologyOutwardEdge`'s

Reply:
It starts to get quite long if we write it out in full `getRightEndpointsForOntologyToOntologyOutwardEdgeKind` :sweat_smile:

Reply:
Why is naming things so hard 😭

Reply:
What do you think about `getRightOntologyEndpointsForOntologyOutwardEdge`?

Reply:
Or `getOntologyEndpointsForOntologyOutwardEdge` (as it's an outward edge)

Reply:
06074e224df3f88dbebcadde3d84b17649b61b07